interfaces/builtin/docker_support.go

```diff
@@ -672,6 +681,43 @@ func (iface *dockerSupportInterface) KModConnectedPlug(spec *kmod.Specification,
 	return nil
 }
 
+func (iface *dockerSupportInterface) AppArmorPermanentPlug(spec *apparmor.Specification, plug *snap.PlugInfo) error {
+	// Generate rules to allow snap-update-ns to mount apparmor config
+	emit := spec.AddUpdateNSf
+	for _, target := range apparmorConfigDirs {
+		base := plug.Snap.Base
+		if base == "" {
+			base = "core"
+		}
+		// unlike below, within the snap mount namespace /snap is
+		// always used
+		source := fmt.Sprintf("/snap/%s/*", base) + target
+		emit("  # access to %s\n", target)
+		emit("  mount options=(bind) %s/ -> %s/,\n", source, target)
+		emit("  umount %s/,\n\n", target)
+	}
+	return nil
+}
+
+func (iface *dockerSupportInterface) MountPermanentPlug(spec *mount.Specification, plug *snap.PlugInfo) error {
+	// preserve apparmor config from the core snap
+	for _, dir := range apparmorConfigDirs {
+		base := plug.Snap.Base
+		if base == "" {
+			base = "core"
+		}
+		// use SnapMountDir since /snap is not always used on the
+		// host mount namespace
+		basePath, _ := filepath.EvalSymlinks(fmt.Sprintf("%s/%s/current", dirs.SnapMountDir, base))
```

Comment:
This is where this falls apart, with fixed error handling you get this:
```
- Setup snap "core" (10126) security profiles (cannot obtain mount security snippets for snap "test-snapd-multipass-support-app": lstat /snap/core/current: no such file or directory)
```

By the time core refreshes it is unlinked and we cannot resolve the symlink.

Reply:
I tried several approaches towards fixing this but I realized they are dead-ends. I will try something different on Monday.

@alexmurray please don't push more stuff here yet.